### ISSUE TITLE
Making NeoVim :help documents NeoVim-specific

### ISSUE THREAD

**Issue by AkamineAiko** [neovim/neovim#22032]
I am reading through the user manual, and there are some Vim-specific instructions/descriptions that might be extremely confusing, especially for newcomers.

For example, in user manual, chapter 02:

https://github.com/neovim/neovim/blob/8144deb0989ea5c61fe9a1a5802d230eba33dfdd/runtime/doc/usr_02.txt#L26-L35

As far as I understand, `gvim` is a command that opens VIM GUI emulator, that has nothing to do with NeoVim.

Then the `vim` command is mentioned multiple times

What is our approach to these files? 

I would suggest editing out those NeoVim-irrelevant things.

The question is - why isn't this user manual "translated" to neovim? Is there something that prevents us to do this?

**Comment by digidoor** [comment]
I'm looking through help files right now to tune up my init.vim, and in the 'p' option of 'guioptions' you find:

> This must be done before starting the GUI.  Set it in your |gvimrc|.  Adding or removing it after the GUI has started has no effect.

I have no idea of nvim even uses the guioptions flags to begin with, but then when there are references to a gvimrc file I just close the help and say "oh well".

**Comment by clason** [comment]
Are you volunteering to rewrite all of the help files, and then making sure we port those updates -- and only those -- that correspond to changes we port from upstream or still apply to Neovim?

**Comment by dundargoc** [comment]
Basically what @clason is saying is that there is not just an initial, one-time cost of changing the documentation to be neovim-only, but an ongoing maintenance cost as any deviation from Vim project makes importing patches harder. If an option straight up doesn't exist then that should be removed from the docs, but if it's just changing "Vim" to "Nvim", then it's probably not worth it. There is no hard rule here unfortunately, the best approach so far has been to evaluate if the benefit of changing is worth the cost of maintenance on a case-by-case basis.

**Comment by AkamineAiko** [comment]
> I'm looking through help files right now to tune up my init.vim, and in the 'p' option of 'guioptions' you find:
> 
> > This must be done before starting the GUI.  Set it in your |gvimrc|.  Adding or removing it after the GUI has started has no effect.
> 
> I have no idea if nvim even uses the guioptions flags to begin with, but then when there are references to a gvimrc file I just close the help and say "oh well".

Yes, exactly those confusing experiences are what I'm talking about here.

I believe there are many things that are straight up irrelevant and should be removed, but I also believe that small little things like changing `vim` to `nvim` in commands should be patched.

I do not really see how changing help files could make importing patches harder, as @dundargoc said.

I'm not volunteering to rewrite all help files, but I will be happy to make large contributions as a part of my learning process, that's basically what I'm doing right now anyways - taking notes or rewriting the user manuals without irrelevant Vim things.

**Comment by AkamineAiko** [comment]
Also, we could make the user manuals more friendly. For example chapter one, for people who just opened neovim and want to learn, get hit with:

https://github.com/neovim/neovim/blob/8144deb0989ea5c61fe9a1a5802d230eba33dfdd/runtime/doc/usr_01.txt#L58-L66

Only then, `:Tutor` command is introduced to them and in chapter 2 `hjkl` movements and other basics are explained.

Why is this `setup-vimrc_example` even there? They are shown multiple commands that I don't even know yet, just to create an EMPTY config file? The concept of vimrc and configuring options is explained **Chapter 5**, and I believe this is where this snippet belongs.  It serves no purpose in chapter 1, it's just confusing.

**Comment by clason** [comment]
I think I need to expand on my (and dundar's) short comment. The documentation is part of the runtime files, which we regularly synchronize with Vim -- this is a core part of the maintenance of Neovim responsible aiming to be as compatible with Vim as is possible. (For example, so you can use nearly all Vim plugins in Neovim.) Vim updates these files in a big batch every couple of weeks, which we need to manually port. This is not a small effort, so any divergence makes this job harder!

For this reason, we treat the upstream documentation -- in particular the manual -- as _Vim's_ documentation; you'll notice that, e.g., `:h eval.txt` explicitly says 
```
VIM REFERENCE MANUAL	  by Bram Moolenaar
```
in the title. This is another reason why we want to make only the minimal necessary changes to avoid blatantly incorrect statements (if we haven't ported a feature yet, or made our own changes).

In contrast, other files specific to Neovim such as `:h lua.txt` say
```
NVIM REFERENCE MANUAL
```
in the title; these files belong completely to us and thus are actively worked on. This was decided to be a better use of our time than touching up Vim's documentation.

Specifically for the user manual (`:h usr_toc.txt`, which should be considered distinct from the reference manual), we have similarly decided that the best option is writing a "Neovim User Manual" from scratch. But this is a _massive_ effort that only makes sense once Neovim has reached 1.0 and the API is stable, but we've taken first steps already with, e.g., `:h lua-guide`.

**Comment by dundargoc** [comment]
@AkamineAiko the example you gave isn't neovim-specific. The best way to improve documentation in those cases is to send a PR to vim. We can then patch that to Neovim afterwards. This helps both projects as well.

**Comment by zeertzjq** [comment]
@dundargoc Vim supports 'compatible', Nvim does not.

**Comment by dundargoc** [comment]
I meant the `setup-vimrc_example` in [this](https://github.com/neovim/neovim/issues/22032#issuecomment-1407653544) comment.

**Comment by zeertzjq** [comment]
How isn't what Neovim-specific?

**Comment by dundargoc** [comment]
My bad, I assumed it did.

I will hide in the shadows and let the pros handle this :)

**Comment by justinmk** [comment]
Improving the docs is an ever-present goal. Rewriting or forking the Vim _user manual_ however is generally out of scope. It's provided as a "nice to have", because it still has some useful parts.

The Nvim _user manual_ is essentially `:help lua-guide`, until further notice.

> Also, we could make the user manuals more friendly.

Some related issues are tracking concrete ideas:


- https://github.com/neovim/neovim/issues/22369
- https://github.com/neovim/neovim/issues/22367
- https://github.com/neovim/neovim/issues/22366
- https://github.com/neovim/neovim/issues/22365
